src/python/WMCore/MicroService/MSRuleCleaner/MSRuleCleanerWflow.py

```diff
@@ -180,6 +180,7 @@ def docSchema(self):
                               'plineAgentCont': False,
                               'plineAgentBlock': False},
             "TransferDone": False  # information - returned by the MSOutput REST call.
+            "TransferTape": False  # information - fetched by Rucio about tape rules completion
```

Comment:
> I'm just trying to decrease the number of flags (checks)

I know, and me myself came with the same idea, but when I tried it, I figured out two things:
   * We take this flag from MSOutput as it is and it is set there by it's own logic, which is: 'Set the transfers to done once we have created the Rucio transfer rules, and do not care about the status of the transfers itself. Which is exactly the opposite of the logic we use here.
   * In case we decide to anyway reuse this flag here, we will end up with having a hidden difference in the logic reflected by the same flag in two different, but related services. And the flag  itself will constantly have different values in those two services - which will only cause confusion and will create situations for faulty debugging and conclusions.

Reply:
That's a valid point. Using the same key name/parameter with different meanings might be confusing indeed. On the other hand, these are different microservices and they have their own object data structure (MSOutputWflow vs MSRuleCleanerWflow), so one can also say their meaning is likely going to be different.

I have no strong preference though. Classical words: whatever is more robust, simple and maintainable ;-)

Reply:
Thanks Alan! in this case I would prefer to keep those flags separated. It also helps in the logging process. It gives clear reason what is the type of showstopper for the workflow to not being able to proceed: either a delay in the MSOutput to create the transfer document and rules or we are in the stage of waiting for a tape rule approval at a site which requires such.